src/Service/Service.php

```diff
@@ -119,6 +119,32 @@ public function jsonSerialize(): array
         return $json;
     }
 
+    /**
+     * @return mixed[]
+     * @throws ServiceException
+     */
+    public function imageJsonSerialize(): array
+    {
+        $this->dockerfileCommands[] = 'FROM ' . $this->image;
```

Comment:
Nope...

If you call the method `imageJsonSerialize()`, you will not get the same results as the state is mutated.
Yet, you want the same results.

You should instead build a temporary array and return that temporary array.